### PR TITLE
Visitor Identification module added.

### PR DESCRIPTION
**Pull request by Sehbaz**
## What to do  
Added visitor identification API code to the HubspotChat component.
Referenced [HubSpot documentation on visitor identification](https://developers.hubspot.com/docs/api/conversation/visitor-identification).

## Background  
The modification was necessary to include the visitor identification API code, which was missing from the library. 
The code has been added based on the information available in the [HubSpot documentation](https://developers.hubspot.com/docs/api/conversation/visitor-identification).

## Acceptance criteria  
The added code for visitor identification functions as intended.
The linting and testing processes pass successfully during GitHub Actions.

### CODE REVIEW

**Comment by Sehbaz** [comment]
@kk-no I have updated the code for the last test fail case, kindly check and approve for test again.

**Comment by kk-no** [comment]
@Sehbaz 
Thank you for the pull request. I re-run the test, but it still seems to fail...
Since it's the end of the year, the review may be a little late, so please bear with me.

**Review by kk-no — COMMENTED** [review]
I made a comment, so please check it as well as fix the test failure.

File: visitor_identification.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+type IdentificationTokenResponse struct {
+	Token string `json:"token"`
+}
+
+// Create creates a Identification Token.
+// In order to bind the created content, a structure must be specified as an argument.
+// When using custom fields, please embed hubspot.request in your own structure.
+func (s *VisitorIdentificationServiceOp) GenerateIdentificationToken(request interface{}) (*IdentificationTokenResponse, error) {
```
[the remark this answers is not in the document]
Comment (Sehbaz) [reply visitor_identification.go]: agreed!

File: gohubspot.go [hunk 2]
```diff
@@ -36,6 +36,8 @@ type Client struct {
 
 	CRM       *CRM
 	Marketing *Marketing
+
+	VisitorIdentification VisitorIdentificationService
```
[the remark this answers is not in the document]
Comment (Sehbaz) [reply gohubspot.go]: Yes, I agree. Code had now been updated

**Comment by Sehbaz** [comment]
@kk-no I have update my code and now that looks far more clean and that should take care of test cases too.

I have made PR ready to be review again.

Thanks.

File: conversation.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package hubspot
+
+type Conversation struct {
+	VisitorIdentification      VisitorIdentificationService
+	IdentificationTokenRequest IdentificationTokenRequest
```
Comment (kk-no) [inline conversation.go:5]: `IdentificationTokenRequest` needs to be switched on a request-by-request basis, so I don't think it's necessary as a field.

File: conversation_visitor_identity.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+	FirstName string `json:"firstname"`
+	LastName  string `json:"lastname"`
+	Email     string `json:"email"`
+}
+
+type VisitorIdentificationService interface {
+	GenerateIdentificationToken(option IdentificationTokenRequest) (*IdentificationTokenResponse, error)
+}
+
+type VisitorIdentificationServiceOp struct {
```
Comment (kk-no) [inline conversation_visitor_identity.go:21]: I want the struct to have a client and an api base path.
(e.g. https://github.com/belong-inc/go-hubspot/blob/main/contact.go#L21)

File: conversation_visitor_identity.go [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+const (
+	visitorIdentificationBasePath = "/conversations/v3/visitor-identification"
+)
+
+type IdentificationTokenResponse struct {
+	Token string `json:"token"`
+}
+
+type IdentificationTokenRequest struct {
+	FirstName string `json:"firstname"`
```
Comment (kk-no) [inline conversation_visitor_identity.go:12]: In the document, it is `firstName`, but it worked with `firstname` at my environment, so there seems to be no problem with the json tag as it is.
https://developers.hubspot.com/docs/api/conversation/visitor-identification

**Review by kk-no — COMMENTED** [review]
Thank you for your update.
It's almost perfect, but I'd like to make the format a little more consistent, so I made a comment. If that is fixed, I think I can merge it!

File: conversation_visitor_identity.go (hunk 5 above)
Comment (Sehbaz) [reply conversation_visitor_identity.go:12]: Updated as requested!

**Comment by Sehbaz** [comment]
@kk-no  I have update my code and I have made PR ready to be review again.

Thank you for valuable feedback.

**Review by kk-no — APPROVED** [review]
LGTM!
We will release `v0.8.0` including this OR in the soon.

**Comment by Sehbaz** [comment]
> LGTM! We will release `v0.8.0` including this OR in the soon.

Amazing, thanks for an update!

**Pull request merged**